Documentation/gettingstarted/clustermesh.rst

```diff
@@ -2,115 +2,259 @@
-in a Kubernetes Secret.
 
-1. Create an etcd configuration file for each remote cluster you want to
-   connect to. The syntax is that the official etcd configuration file and
-   identical to the syntax used in the ``cilium-config`` ConfigMap.
+.. note::
+
+   Make sure that you create the service in namespace in which cilium and/or
+   etcd is running. Depending on which installation method you chose, this
+   could be ``kube-system`` or ``cilium``.
+
+Extract the TLS keys and generate the etcd configuration
+========================================================
+
+The clustermesh control plane performs TLS based authentication and encryption.
+For this purpose, the TLS keys and certificates of each etcd need to be made
+available to all clusters that wish to connect.
+
+1. Clone the ``cilium/clustermesh-tools`` repository. It contains scripts to
+   extracts the secrets and generate a Kubernetes secret in form of a YAML
+   file:
+
+   .. code:: bash
+   
+        git clone https://github.com/cilium/clustermesh-tools.git
+        cd clustermesh-tools
+
+2. Ensure that the kubectl context is pointing to the cluster you want to
+   extract the secret from.
+
+3. Extract the TLS certificate, key and root CA authority.
+
+   .. code:: bash
+
+        ./extract-etcd-secrets.sh
+
+   This will extract the keys that Cilium is using to connect to the etcd in
+   the local cluster. The key files are written to
+   ``config/<cluster-name>.*.{key|crt|-ca.crt}``
 
-2. Create a secret ``cilium-clustermesh`` from all configuration files you have
-   created:
+4. Repeat this step for all clusters you want to connect with each other.
+
+5. Generate a single Kubernetes secret from all the keys and certificates
+   extracted. The secret will contain the etcd configuration with the service
+   IP or host name of the etcd including the keys and certificates to access
+   it.
 
    .. code:: bash
 
-       $ ks create secret generic cilium-clustermesh --from-file=./cluster5 --from-file=./cluster7
+        ./generate-secret-yaml.sh > clustermesh.yaml
+
+.. note::
+
+   The key files in ``config/`` and the secret represented as YAML are
+   sensitive. Anyone gaining access to these files is able to connect to the
+   etcd instances in the local cluster. Delete the files after the you are done
+   setting up the clustermesh.
+
+Ensure that the etcd service names can be resolved
+==================================================
+
+For TLS authentication to work properly, agents will connect to etcd in remote
+clusters using a pre-defined naming schema ``{clustername}.mesh.cilium.io``. In
+order for DNS resolution to work on these virtual host name, the names are
+statically mapped to the service IP via the ``/etc/hosts`` file.
+
+1. The following script will generate the required segment which has to be
+   inserted into the ``cilium`` DaemonSet:
+
+    .. code:: bash
+
+       ./generate-name-mapping.sh
```

Comment:
replace these steps with:
```
kubectl patch ds -n kube-system cilium --patch "$(./generate-name-mapping.sh)"
```
where the output of `./generate-name-mapping.sh` should be:

```
spec:
  template:
    spec:
      hostAliases:
      - hostnames:
        - cluster1.mesh.cilium.io
        ip: 10.138.0.18
      - hostnames:
        - cluster2.mesh.cilium.io
        ip: 10.138.0.19
```